lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexer.java

```diff
@@ -65,20 +92,61 @@ public boolean validateConnection() {
 
   @Override
   protected void sendToIndex(List<Document> documents) throws IndexerException {
-    if (namespaces != null) {
-      for (Map.Entry<String, Object> entry : namespaces.entrySet()) {
-        uploadDocuments(documents, (String) entry.getValue(), entry.getKey());
+    // retrieve documents to delete & upload
+    List<Document> deleteList = new ArrayList<>();
+    List<Document> uploadList = new ArrayList<>();
+    for (Document doc : documents) {
+      if (isDeletion(doc)) {
+        // if doc is already in uploadList and is now marked for deletion, then remove that document in upload
+        if (uploadList.contains(doc)) {
```

Comment:
The contains(doc), remove(doc) won't work as expected here due to relying on Object.equals.

Reply:
You can use something like uploadList.filter(other -> other.id.equals(doc.id));